avocado/core/test.py

```diff
@@ -1421,12 +1421,28 @@ class DryRunTest(MockingTest):
     Fake test which logs itself and reports as CANCEL
     """
 
+    def __init__(self, *args, **kwargs):
+        if 'modulePath' in kwargs:
```

Comment:
It looks like the `modulePath` key can be present, but set to None.  Reproducer is:

```
avocado --show=test run --dry-run -- /bin/true
```

Which results in:

```
...
[stderr]   File "/home/cleber/src/avocado/avocado/avocado/core/test.py", line 426, in __init__
[stderr]     if self.filename:
[stderr]   File "/home/cleber/src/avocado/avocado/avocado/core/test.py", line 1438, in filename
[stderr]     source = os.path.abspath(self._filename)
[stderr]   File "/usr/lib64/python3.7/posixpath.py", line 378, in abspath
[stderr]     path = os.fspath(path)
[stderr] TypeError: expected str, bytes or os.PathLike object, not NoneType
```

Reply:
This didn't come to my mind. I will look at that, thanks